test/test_torch.py

```diff
@@ -18181,6 +18181,57 @@ def test_large_linspace(self, device, dtype):
         x = torch.linspace(start, end, steps, dtype=dtype, device=device)
         self.assertGreater(x[1] - x[0], (end - start) / steps)
 
+    @dtypes(torch.int64, torch.float, torch.complex128)
+    def test_movedim_invalid(self, device, dtype):
+        shape = self._rand_shape(4, min_size=5, max_size=10)
+        x = self._generate_input(shape, dtype, device, False)
+
+        # Invalid `src` and `dst` dimension
+        with self.assertRaisesRegex(IndexError, "Dimension out of range"):
+            torch.movedim(x, 5, 0)
+
+        with self.assertRaisesRegex(IndexError, "Dimension out of range"):
+            torch.movedim(x, 0, 5)
+
+        # Mismatch in size of `src` and `dst`
+        with self.assertRaisesRegex(RuntimeError, "movedim: Invalid source or destination dims:"):
+            torch.movedim(x, (1, 0), (0, ))
+
+        with self.assertRaisesRegex(RuntimeError, "movedim: repeated dim in `src`"):
+            torch.movedim(x, (0, 0), (0, 1))
+
+        with self.assertRaisesRegex(RuntimeError, "movedim: repeated dim in `dst`"):
+            torch.movedim(x, (0, 1), (1, 1))
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(torch.int64, torch.float, torch.complex128)
+    def test_movedim(self, device, dtype):
+        for nd in range(5):
+            shape = self._rand_shape(nd, min_size=5, max_size=10)
+            x = self._generate_input(shape, dtype, device, with_extremal=False)
+            for random_negative in [True, False]:
+                for src_dim, dst_dim in permutations(range(nd), r=2):
+                    if random_negative:
+                        src_dim = src_dim - nd
```

Comment:
This is called `random_negative`, but... it's not really random in this case. The `src` dim becomes either negative, or not negative. I don't see a test for where we move a single dimension and where the `dst` dim is negative.

Reply:
> I don't see a test for where we move a single dimension
```python
                for src_dim, dst_dim in permutations(range(nd), r=2):
                    if random_negative:
                        src_dim = src_dim - nd
                    # Integer Inputs
                    torch_fn = partial(torch.movedim, src=src_dim, dst=dst_dim)
                    np_fn = partial(np.moveaxis, source=src_dim, destination=dst_dim)
                    self.compare_with_numpy(torch_fn, np_fn, x, device=None, dtype=None)
```

The one labelled with `Integer Inputs` actually tests for single dimension case.

```python
>>> for src, dst in (permutations(range(3), r=2)):
           print(src, dst)
(0, 1)
(0, 2)
(1, 0)
(1, 2)
(2, 0)
(2, 1)
```

Will add a case when `dst` dim is negative.